### PR TITLE
Add some documentation to the various library code

### PR DESCRIPTION
**Pull request by teopiaz**

### CODE REVIEW

**Comment by teopiaz** [comment]
force push to fixed the conflicts due to the removal of the addresses from the function names.

**Comment by Jayveer** [comment]
Personally I feel like we shouldn't change the original names for variables/functions that we already know

**Comment by teopiaz** [comment]
> Personally I feel like we shouldn't change the original names for variables/functions that we already know

HikituriFlag I could revert if you prefer, but GV_PassageTime_800AB924 doesn't have much meaning. Delta Time is what usually is used in this context.

**Comment by KieronJ** [comment]
GV_PassageTime is an original name, but I agree it is terrible.

**Comment by GirianSeed** [comment]
> Personally I feel like we shouldn't change the original names for variables/functions that we already know

I second this but I'm 100% for at least commenting the real meaning/use of these things.

**Comment by teopiaz** [comment]
Thank you for the feedback. I will fix this patch.

File: src/libgv/actor.c [hunk 1]
```diff
@@ -201,13 +260,20 @@ void GV_InitActor(int level, GV_ACT *actor, TActorFreeFunction free_func)
...
 
-GV_ACT *GV_NewActor(int level, int memSize)
+/**
+ * @brief Allocate memory for an actor and initialize it.
+ *
+ * @param actorListId The id of the list where the actor will be added.
+ * @param memSize The size of the actor.
+ * @return GV_ACT* The allocated actor.
+ */
+GV_ACT *GV_NewActor(int actorListId, int memSize)
```
Comment (GirianSeed) [inline src/libgv/actor.c:270]: Actors pass a macro called EXEC_LEVEL to GV_NewActor. Calling this actorListId implies something different imo. The actual list is supposed to be a private implementation detail in actor.c anyway.

Comment (teopiaz) [reply src/libgv/actor.c:270]: I will rename to execLevel

**Comment by KieronJ** [comment]
Looks good, thanks!

**Pull request merged**